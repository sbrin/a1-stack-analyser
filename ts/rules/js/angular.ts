import { register } from '../../register.js';

register({
  tech: 'angular',
  name: 'Angular',
  type: 'language',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
