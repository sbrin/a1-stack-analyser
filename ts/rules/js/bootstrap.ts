import { register } from '../../register.js';

register({
  tech: 'bootstrap',
  name: 'Bootstrap',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'bootstrap' }],
});
