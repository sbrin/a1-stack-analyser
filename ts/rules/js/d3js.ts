import { register } from '../../register.js';

register({
  tech: 'd3js',
  name: 'D3.js',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'd3' },
    { type: 'npm', name: 'd3-shape' },
    { type: 'npm', name: '@types/d3' },
    { type: 'npm', name: '@types/d3-shape' },
  ],
});
