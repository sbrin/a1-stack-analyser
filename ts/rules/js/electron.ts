import { register } from '../../register.js';

register({
  tech: 'electron',
  name: 'Electron',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'electron' },
    { type: 'githubAction', name: 'samuelmeuli/action-electron-builder' },
  ],
});
