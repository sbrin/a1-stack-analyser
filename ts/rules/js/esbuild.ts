import { register } from '../../register.js';

register({
  tech: 'esbuild',
  name: 'Esbuild',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'esbuild' }],
});
