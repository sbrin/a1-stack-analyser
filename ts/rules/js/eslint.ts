import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: ['.eslintrc', '.eslintrc.cjs', '.eslintrc.json', '.eslintrc.js'],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
