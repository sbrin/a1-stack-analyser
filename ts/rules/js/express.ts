import { register } from '../../register.js';

register({
  tech: 'express',
  name: 'Express',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'express' }],
});
