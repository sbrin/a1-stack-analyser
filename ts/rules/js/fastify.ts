import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'fastify' },
    { type: 'npm', name: '@opentelemetry/instrumentation-fastify' },
  ],
});
