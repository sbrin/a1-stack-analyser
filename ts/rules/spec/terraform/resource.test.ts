import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import type { AllowedKeys } from '../../../index.js';
import { rawList } from '../../../loader.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';

import '../../../autoload.js';

describe('terraform (resource)', () => {
  it('should match everything', async () => {
    const resource: string[] = [``];

    for (const item of rawList) {
      if (
        item.type !== 'dependency' ||
        item.ref.type !== 'terraform.resource'
      ) {
        continue;
      }

      const example = 'example' in item.ref ? item.ref.example : item.ref.name;
      resource.push(`
      resource "${example}" "foobar" {
        name     = "hello"
      }
    `);
    }

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['main.tf'],
        },
        files: {
          '/main.tf': resource.join(''),
        },
      }),
    });

    expect(
      Array.from(flatten(res, { merge: true }).techs).sort()
    ).toMatchSnapshot();
  });

  it('should match in tf file', async () => {
    const resource: string[] = [
      `
    resource "unknown" "foobar" {
      name     = "hello"
    }`,
    ];

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['main.tf'],
        },
        files: {
          '/main.tf': resource.join(''),
        },
      }),
    });

    const match: AllowedKeys[] = ['terraform'];
    const merged = flatten(res, { merge: true });
    expect(Array.from(merged.techs).sort()).toStrictEqual(match);
  });

  it('should match nothing but register dependencies', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['.terraform.lock.hcl'],
        },
        files: {
          '/.terraform.lock.hcl': `
          provider "helloworld" {
            version = "0.0.0"
          }
          `,
        },
      }),
    });

    const match: AllowedKeys[] = ['terraform'];
    const merged = flatten(res, { merge: true });
    expect(Array.from(merged.techs).sort()).toStrictEqual(match);
    expect(Array.from(merged.dependencies).sort()).toMatchSnapshot();
  });
});
