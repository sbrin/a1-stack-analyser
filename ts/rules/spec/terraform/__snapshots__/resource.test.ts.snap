// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`terraform (resource) > should match everything 1`] = `
[
  "atlassian",
  "atlassian.jira",
  "aws",
  "aws.amplifyhosting",
  "aws.apigateway",
  "aws.athena",
  "aws.cloudformation",
  "aws.cloudfront",
  "aws.cloudsearch",
  "aws.cloudwatch",
  "aws.codebuild",
  "aws.documentdb",
  "aws.dynamodb",
  "aws.ebs",
  "aws.ec2",
  "aws.ecr",
  "aws.ecs",
  "aws.efs",
  "aws.eks",
  "aws.elasticache",
  "aws.glacier",
  "aws.glue",
  "aws.kafka",
  "aws.kinesis",
  "aws.kms",
  "aws.lambda",
  "aws.lightsail",
  "aws.memorydb",
  "aws.mq",
  "aws.neptune",
  "aws.opensearch",
  "aws.rds",
  "aws.redshift",
  "aws.s3",
  "aws.sagemaker",
  "aws.secretsmanager",
  "aws.ses",
  "aws.sfn",
  "aws.sns",
  "aws.sqs",
  "aws.timestream",
  "azure",
  "azure.aks",
  "azure.cosmosdb",
  "azure.functions",
  "azure.mariadb",
  "azure.mysql",
  "azure.postgres",
  "azure.redis",
  "azure.sql",
  "azure.staticwebapps",
  "azure.storage",
  "cloudflare",
  "cloudflare.pages",
  "cloudflare.workers",
  "datadog",
  "firebase",
  "firebase.firestore",
  "gcp",
  "gcp.aiplatform",
  "gcp.appengine",
  "gcp.artifactregistry",
  "gcp.bigquery",
  "gcp.bigtable",
  "gcp.cloudbuild",
  "gcp.cloudrun",
  "gcp.dataflow",
  "gcp.dataproc",
  "gcp.datastore",
  "gcp.dialogflow",
  "gcp.dns",
  "gcp.functions",
  "gcp.gce",
  "gcp.gcs",
  "gcp.gke",
  "gcp.kms",
  "gcp.logging",
  "gcp.memorystore",
  "gcp.pubsub",
  "gcp.secretmanager",
  "gcp.spanner",
  "gcp.sql",
  "gcp.tasks",
  "ovh",
  "ovh.database",
  "ovh.dedicated",
  "ovh.kubernetes",
  "ovh.storage",
  "ovh.vps",
  "qovery",
  "qovery.cluster",
  "qovery.database",
  "scaleway",
  "scaleway.container",
  "scaleway.database",
  "scaleway.documentdb",
  "scaleway.elasticmetal",
  "scaleway.function",
  "scaleway.kubernetes",
  "scaleway.mq",
  "scaleway.redis",
  "scaleway.secretmanager",
  "scaleway.storage",
  "terraform",
]
`;

exports[`terraform (resource) > should match nothing but register dependencies 1`] = `
[
  [
    "terraform",
    "helloworld",
    "0.0.0",
  ],
]
`;
